docs/fastdds/dds_layer/core/policy/standardQosPolicies.rst

```diff
@@ -463,24 +472,48 @@ There are two possible values (see |HistoryQosPolicyKind-api|):
   BEST_EFFORT the older values will be discarded, but if it is RELIABLE the service blocks the DataWriter until the old
   values are delivered to all existing Subscribers.
 
+.. _historyqospolicy_consistency:
+
+Consistency rule
+""""""""""""""""
+
+The HistoryQos must be set consistently with the :ref:`resourcelimitsqospolicy`, but also other QoS as
+:ref:`durabilityqospolicy`, so there are several cases to take into account:
+
+* The |HistoryQosPolicy::depth-api| must be consistent with the :ref:`resourcelimitsqospolicy` settings, which means
+  that the |HistoryQosPolicy::depth-api| must be equal or lower than the :ref:`resourcelimitsqospolicy`'s
+  |ResourceLimitsQosPolicy::max_samples_per_instance-api|.
+  Also, |ResourceLimitsQosPolicy::max_samples-api| must be equal or higher than the product of
+  |ResourceLimitsQosPolicy::max_samples_per_instance-api| and |ResourceLimitsQosPolicy::max_instances-api|.
+* The |HistoryQosPolicy::depth-api| cannot be zero.
```

Comment:
We may start here stating that the `depth` is only taken into consideration when `kind` is `KEEP_LAST`. Also, when we finally agree on the `Fast DDS's PR` how we manage having a LENGTH_UNLIMITED, we update it here.

Reply:
Finally, we should state here that `depth` cannot be less or equal than 0.